reposerver/repository/repository.go

```diff
@@ -488,3 +499,29 @@ func (s *Service) newClientResolveRevision(repo *v1alpha1.Repository, revision s
 	}
 	return gitClient, commitSHA, nil
 }
+
+func runCustomTool(appPath string, q *ManifestRequest, tools []*v1alpha1.TemplatingTool) ([]*unstructured.Unstructured, error) {
+	var tool *v1alpha1.TemplatingTool
+	for i := range tools {
+		if tools[i].Name == q.ApplicationSource.TemplatingTool.Name {
+			tool = tools[i]
+			break
+		}
+	}
+	if tool == nil {
+		return nil, fmt.Errorf("Custom templating tool with name '%s' is not supported.", q.ApplicationSource.TemplatingTool.Name)
+	}
+	env := []string{"ARGOCD_APP=" + q.AppLabelValue, "ARGOCD_NAMESPACE=" + q.Namespace}
```

Comment:
1. I think we should namespace these env vars, similar to their metadata field names:
```
ARGOCD_APP_NAME
ARGOCD_APP_NAMESPACE
```
In the future it will let us do things like:
```
ARGOCD_APP_CLUSTER
ARGOCD_APP_DIRECTORY_BASENAME
```

2. Can we add these env vars as constants somewhere?
```
PluginEnvAppName = ARGOCD_APP_NAME
PluginEnvAppNamespace = ARGOCD_APP_NAMESPACE
```

3. Do we inherit the environment variables of the system? If not I think we need to.

Reply:
done